internal/core/item_tree.rs

```diff
@@ -184,6 +185,10 @@ impl ItemRc {
         self.borrow().as_ref().geometry()
     }
 
+    pub fn logical_geometry(&self) -> LogicalRect {
+        LogicalRect::from_untyped(&self.borrow().as_ref().geometry())
```

Comment:
geometry()  could return a LogicalRect

Reply:
Yes. If we do that, then we also need to change all our internal event handling across. I thought of this as a separate change.